vm/src/utils.rs

```diff
@@ -109,3 +110,19 @@ impl IntoPyObject for std::convert::Infallible {
         match self {}
     }
 }
+
+pub trait IntoCString {
+    fn to_cstring(&self, vm: &VirtualMachine) -> PyResult<std::ffi::CString>;
+}
+
+impl IntoCString for &str {
+    fn to_cstring(&self, vm: &VirtualMachine) -> PyResult<std::ffi::CString> {
+        std::ffi::CString::new(*self).map_err(|err| err.into_pyexception(vm))
+    }
+}
+
+impl IntoCString for PyStr {
+    fn to_cstring(&self, vm: &VirtualMachine) -> PyResult<std::ffi::CString> {
+        std::ffi::CString::new(self.as_ref()).map_err(|err| err.into_pyexception(vm))
```

Comment:
The naming convention for error-returning traits are `TryFrom` and `TryInto` in Rust. In Rust, `to` prefix and `into` prefix have different meaning, so the method name must use `into` prefix. (create vs move)

I think `self.as_ref().try_into_cstring(vm)` will work here.

I originally thought just adding a free function will be enough, but this version also look great.

Reply:
Ok Thanks! I applied it 😊

By the way, I am considering the way for removing duplicated code in both PyStr and &str.
```rust
std::ffi::CString::new(self.as_ref()).map_err(|err| err.into_pyexception(vm)) // PyStr impl
std::ffi::CString::new(*self).map_err(|err| err.into_pyexception(vm)) // &str impl
```

I found that PyStr and str both implementing [`AsRef`](https://doc.rust-lang.org/std/convert/trait.AsRef.html) trait and I want to utilize it something like below.
```rust
pub trait ToCString<T: 'static + ?Sized> : AsRef<T> {
    fn to_cstring(&self, vm: &VirtualMachine) -> PyResult<std::ffi::CString> {
        std::ffi::CString::new(self.as_ref()).map_err(|err| err.into_pyexception(vm))
    }
}

impl ToCString<str> for str {}
impl ToCString<str> for PyStr {}
```

But I cant resolve compile error.. Could you give me any hint for solving this? Thanks!